Add event callbacks on command handlers

MariuszSkamra opened the pull request:
Those callbacks will be needed to i.e. track state of firmware update,
Notify about reboot going to happen etc.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

nvlsianpu commented on cmd/img_mgmt/include/img_mgmt/img_mgmt.h:
```diff
@@ -37,6 +37,13 @@ extern "C" {
+/*
+ * IMG_MGMT statuses.
+ */
+#define IMG_MGMT_STATUS_UPLOAD_START            0x80
```
Why starts from 0x80? I can expect that this is related to MGMT_STATUS_CMD_xxx which will be added to the result status.

MariuszSkamra replied:
Exactly, this is due to MGTM_STATUS* codes that are 0x01 and 0x02.

rymanluk commented on cmd/img_mgmt/src/img_mgmt.c:
```diff
@@ -424,11 +425,15 @@ img_mgmt_upload(struct mgmt_ctxt *ctxt)
             return MGMT_ERR_EINVAL;
         }
 
+        status = IMG_MGMT_STATUS_UPLOAD_ONGOING;
```
Did you consider error status? Once it is set as ONGOING, below I can find couple of places where we could leave without a success.

MariuszSkamra replied:
I think, I overlooked this one. I will double check it, Thanks!

MariuszSkamra commented:
I updated PR with respect to your commants
@rymanluk everytime, command will fail, MGMT_EVT_OP_CMD_DONE will be sent anyway, with error status as defined by MGMT_ERR_* defines.
@nvlsianpu To avoid confusion, I changed statuses to opcodes MGMT_EVT_OP_CMD_* adding 3rd MGMT_EVT_OP_CMD_STATUS which argument can be status of upload. This way I avoid 0x80 shift, which might be confusing.
@carlescufi I will create an issue for this one to keep it in mind. Note, that it is not necessary feature, and in some cases (lack of display??) percentages are not needed IMO.

nvlsianpu commented on cmd/img_mgmt/src/img_mgmt.c:
```diff
@@ -341,6 +341,7 @@ img_mgmt_upload_first_chunk(struct mgmt_ctxt *ctxt, const uint8_t *req_data,
 static int
 img_mgmt_upload(struct mgmt_ctxt *ctxt)
 {
+    struct mgmt_evt_op_cmd_status_arg arg;
```
name to generic (as this is long routine): prefer `status_arg` or cmd_status_arg

MariuszSkamra commented:
@nvlsianpu done
